# force libuv to use the older fork method rather than the newer spawn method
# of process execution on older systems. Although older systems 10.5 and newer do
# support posix_spawn, the implementation used in libuv is not working well on these
# older systems. 
# see https://trac.macports.org/ticket/67540
# see https://trac.macports.org/ticket/68718
# and related issues referenced in those tickets

# and add some minor older system tweaks needed from the libuv port

--- Utilities/cmlibuv/src/unix/process.c.orig
+++ Utilities/cmlibuv/src/unix/process.c
@@ -37,7 +37,10 @@
 #include <sched.h>
 
 #if defined(__APPLE__)
+# include <AvailabilityMacros.h>
+# if MAC_OS_X_VERSION_MAX_ALLOWED >= 1050
 # include <spawn.h>
+# endif
 # include <paths.h>
 # include <sys/kauth.h>
 # include <sys/types.h>
@@ -428,7 +431,7 @@
 #endif
 
 
-#if defined(__APPLE__)
+#if defined(__APPLE__) && (MAC_OS_X_VERSION_MIN_REQUIRED >= 1080)
 typedef struct uv__posix_spawn_fncs_tag {
   struct {
     int (*addchdir_np)(const posix_spawn_file_actions_t *, const char *);
@@ -880,7 +885,7 @@
   int exec_errorno;
   ssize_t r;
 
-#if defined(__APPLE__)
+#if defined(__APPLE__) && (MAC_OS_X_VERSION_MIN_REQUIRED >= 1080)
   uv_once(&posix_spawn_init_once, uv__spawn_init_posix_spawn);
 
   /* Special child process spawn case for macOS Big Sur (11.0) onwards


--- ./Utilities/cmlibuv/src/unix/udp.c.orig
+++ ./Utilities/cmlibuv/src/unix/udp.c
@@ -944,6 +944,7 @@
     !defined(__ANDROID__) &&                                        \
     !defined(__DragonFly__) &&                                      \
     !defined(__QNX__) &&                                            \
+    (!defined(__APPLE__) || (MAC_OS_X_VERSION_MAX_ALLOWED >= 1070)) && \
     !defined(__GNU__)
 static int uv__udp_set_source_membership4(uv_udp_t* handle,
                                           const struct sockaddr_in* multicast_addr,
@@ -1137,6 +1138,7 @@
     !defined(__ANDROID__) &&                                        \
     !defined(__DragonFly__) &&                                      \
     !defined(__QNX__) &&                                            \
+    (!defined(__APPLE__) || (MAC_OS_X_VERSION_MAX_ALLOWED >= 1070)) && \
     !defined(__GNU__)
   int err;
   union uv__sockaddr mcast_addr;
